tools/src/commands/GenerateBareApp.ts

```diff
@@ -90,9 +92,25 @@ async function createProjectDirectory({
   localTemplate: boolean;
 }) {
   if (localTemplate) {
+    // For local template, create an NPM package from the local template, then unpack it
+    // using the same transforms that @expo/cli uses
     const pathToBareTemplate = path.resolve(EXPO_DIR, 'templates', 'expo-template-bare-minimum');
-    const pathToWorkspace = path.resolve(workspaceDir, appName);
-    return fs.copy(pathToBareTemplate, pathToWorkspace, { recursive: true });
+    const pathToAppDirectory = path.resolve(workspaceDir, appName);
+    const templateVersion = require(path.join(pathToBareTemplate, 'package.json')).version;
+    await spawnAsync('npm', ['pack', '--pack-destination', workspaceDir], {
```

Comment:
Maybe you could use the existing similar implementation in `Npm.ts`, see https://github.com/expo/expo/blob/ce17bd92ffe5923c9c1f8675e7b031e9e044443b/tools/src/Npm.ts#L53

Reply:
Sadly that doesn't do the name transform.

Reply:
Is the name transform really necessary?

Reply:
Yes it is, otherwise when you build the app, it will attempt to load HelloWorld from the AppRegistry instead of the correct app name -- and there will be many other problems too.